threedi_modelchecker/migrations/versions/0206_control_structures.py

```diff
@@ -0,0 +1,51 @@
+"""make settings nullable
+
+Revision ID: 0206
+Revises: 0205
+Create Date: 2021-12-16 13:30:00
+
+"""
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = "0206"
+down_revision = "0205"
+branch_labels = None
+depends_on = None
+
+
+MIGRATION_QUERIES = """
+UPDATE v2_control_measure_map SET weight = 1.0 WHERE weight IS NULL;
+"""
+
+
+def upgrade():
+    for q in MIGRATION_QUERIES.split(";"):
+        op.execute(q)
+
+    with op.batch_alter_table("v2_control_measure_map") as batch_op:
+        batch_op.alter_column("object_id", nullable=False)
```

Comment:
hmmm this is not backwards compatible, however I think these fields are actually mandatory, so should have nullable=false.

Reply:
Yes all 1800 inpy-success models have no NULL values in these columns. The only one is the "weight" colom for which I added a data migration.

I could (as a safety precaution) do queries like:  `DELETE FROM v2_control_measure_map WHERE object_id IS NULL OR object_type IS NULL OR ...` so that migrations never fail.